cmd/private-org-peribolos-sync/main.go

```diff
@@ -0,0 +1,254 @@
+	fs.StringVar(&o.releaseRepoPath, "release-repo-path", "", "Path to a openshift/release repository directory")
+	fs.StringVar(&o.destOrg, "destination-org", "", "Destination name of the peribolos configuration organzation")
+
+	fs.Var(&o.includeRepos, "include-repo", "One or more repositories that do not promote official images and need to be included")
+
+	o.github.AddFlags(fs)
+
+	args := os.Args[1:]
+	if err := fs.Parse(args); err != nil {
+		os.Exit(1)
+	}
+	fs.Parse(args)
+
+	return o
+}
+
+func validateOptions(o *options) error {
+	var validationErrors []error
+
+	if len(o.releaseRepoPath) == 0 {
+		validationErrors = append(validationErrors, errors.New("--release-repo-path is not specified"))
+	}
+	if len(o.peribolosConfig) == 0 {
+		validationErrors = append(validationErrors, errors.New("--peribolos-config is not specified"))
+	}
+	if len(o.destOrg) == 0 {
+		validationErrors = append(validationErrors, errors.New("--destination-org is not specified"))
+	}
+
+	if len(o.includeRepos.values) > 0 {
+		o.includeReposSet = make(map[string]sets.String)
+		for _, repo := range o.includeRepos.values {
+			orgRepo := strings.Split(repo, "/")
+			if len(orgRepo) != 2 {
+				validationErrors = append(validationErrors, fmt.Errorf("--include-repo: %s must be in org/repo format", repo))
+			} else {
+				if _, ok := o.includeReposSet[orgRepo[0]]; !ok {
+					o.includeReposSet[orgRepo[0]] = sets.NewString([]string{orgRepo[1]}...)
+				} else {
+					o.includeReposSet[orgRepo[0]].Insert(orgRepo[1])
+				}
+			}
+		}
+	}
+
+	if err := o.github.Validate(false); err != nil {
+		validationErrors = append(validationErrors, err)
+	}
+
+	return kerrors.NewAggregate(validationErrors)
+}
+
+func main() {
+	o := gatherOptions()
+	err := validateOptions(&o)
+	if err != nil {
+		logrus.WithError(err).Fatal("invalid options")
+	}
+
+	go func() {
+		interrupts.WaitForGracefulShutdown()
+		os.Exit(1)
+	}()
+
+	logger := logrus.WithField("destination-org", o.destOrg)
+
+	b, err := ioutil.ReadFile(o.peribolosConfig)
+	if err != nil {
+		logger.WithError(err).Fatal("could not read peribolos configuration file")
+	}
+
+	var peribolosConfig org.FullConfig
+	if err := yaml.Unmarshal(b, &peribolosConfig); err != nil {
+		logger.WithError(err).Fatal("failed to unmarshal peribolos config")
+	}
+
+	secretAgent := &secret.Agent{}
+	if err := secretAgent.Start([]string{o.github.TokenPath}); err != nil {
+		logrus.WithError(err).Fatal("Error starting secrets agent.")
+	}
+	gc, err := o.github.GitHubClient(secretAgent, false)
+	if err != nil {
+		logger.WithError(err).Fatal("Error getting GitHub client.")
+	}
+
+	orgRepos, err := getOrgReposWithOfficialImages(o.releaseRepoPath, o.includeReposSet)
+	if err != nil {
+		logger.WithError(err).Fatal("couldn't get the list of org/repos that promote official images")
+	}
+
+	peribolosRepos := generateRepositories(gc, orgRepos, logger)
+	peribolosConfigByOrg := peribolosConfig.Orgs[o.destOrg]
```

Comment:
Something is fishy about this ... why not

```go
peribolosConfig.Orgs[o.destOrg].Repos = peribolosRepos
```

Reply:
It seems that golang doesn't let us do that.